File: lib/models/product.dart
// import 'dart:convert';

// class Product {
//   final double? discountPrice;
//   final String? offer;
//   final String subCategory;
//   final String name;
//   final String description;
//   final double? quantity;  
//   final List<String> images;
//   // final List<String> scheduleOptions;
//   final String category;
//   final double price;
//   final String schedule;
//   final String? id;
  
//   Product({
//     required this.discountPrice,
//     required this.offer,
//     required this.subCategory,
//     required this.name,
//     required this.description,
//     required this.quantity,
//     required this.images,
//     // required this.scheduleOptions,
//     required this.category,
//     required this.price,
//     required this.schedule,
//     this.id,
//   });

//   Map<String, dynamic> toMap() {
//     return <String, dynamic>{
//       'discountPrice': discountPrice,
//       // 'scheduleOptions': scheduleOptions,
//       'schedule' : schedule,
//       'offer' : offer,
//       'subCategory': subCategory,
//       'name': name,
//       'description': description,
//       'quantity': quantity,
//       'images': images,
//       'category': category,
//       'price': price,
//       'id': id,
//     };
//   }

//   factory Product.fromMap(Map<String, dynamic> map) {
//     return Product(
//       discountPrice: map['discountPrice']?.toDouble() ?? 0.0,
//       offer: map['offer'] ?? '',
//       subCategory: map['subCategory'] ?? '',
//       name: map['name'] ?? '',
//       description: map['description'] ?? '',
//       quantity: map['quantity']?.toDouble() ?? 0.0,
//       images: List<String>.from(map['images']),
//       // scheduleOptions: List<String>.from(map['scheduleOptions']),
//       category: map['category'] ?? '',
//       price: map['price']?.toDouble() ?? 0.0,
//       schedule: map['schedule'] ?? '',
//       id: map['_id'],
//     );
//   }

//   String toJson() => json.encode(toMap());

//   factory Product.fromJson(String source) => Product.fromMap(json.decode(source) as Map<String, dynamic>);
// }



import 'dart:convert';

class Product {
  final double? discountPrice;
  final String? offer;
  final String subCategory;
  final String name;
  final String description;
  final double? quantity;
  final List<String> images;
  final String category;
  final double price;
  final String schedule;
  final String? id;

  Product({
    required this.discountPrice,
    required this.offer,
    required this.subCategory,
    required this.name,
    required this.description,
    required this.quantity,
    required this.images,
    required this.category,
    required this.price,
    required this.schedule,
    this.id,
  });

  Map<String, dynamic> toMap() {
    return {
      'discountPrice': discountPrice,
      'offer': offer,
      'subCategory': subCategory,
      'name': name,
      'description': description,
      'quantity': quantity,
      'images': images,
      'category': category,
      'price': price,
      'schedule': schedule,
      '_id': id,
    };
  }

  factory Product.fromMap(Map<String, dynamic> map) {
    return Product(
      discountPrice: map['discountPrice']?.toDouble(),
      offer: map['offer'],
      subCategory: map['subCategory'] ?? '',
      name: map['name'] ?? '',
      description: map['description'] ?? '',
      quantity: map['quantity']?.toDouble(),
      images: List<String>.from(map['images']),
      category: map['category'] ?? '',
      price: map['price']?.toDouble() ?? 0.0,
      schedule: map['schedule'] ?? '',
      id: map['_id'],
    );
  }

  String toJson() => json.encode(toMap());

  factory Product.fromJson(String source) => Product.fromMap(json.decode(source));
}

File: server/models/product.js
// const mongoose = require("mongoose");


// const productSchema = mongoose.Schema({
//   shopName: {
//     type: String,
//     // required: true,
//     trim: true,
//   },

//   shopDescription: {
//     type: String,
//     // required: true,
//     trim: true,
//   },

//   shopId:{
//     type: String,
//     // required:true,
//     trim:true,
//   },

//   productName: {
//     type: String,
//     // required: true,
//     trim: true,
//   },

  
//   productDescription: {
//     type: String,
//     // required: true,
//     trim: true,
//   },
  
//   images: [
//     {
//       type: String,
//       required: true,
//     },
//   ],
//   quantity: {
//     type: String,
//     required: true,
//   },
//   price: {
//     type: Number,
//     required: true,
//   },
//   category: {
//     type: String,
//     required: true,
//   },
// //   ratings: [ratingSchema],
// });

// const Product = mongoose.model("Product", productSchema);
// module.exports = { Product, productSchema };


const mongoose = require("mongoose");


const productSchema = mongoose.Schema({
  name: {
    type: String,
    required: true,
    trim: true,
  },

  description: {
    type: String,
    required: true,
    trim: true,
  },

  category:{
    type: String,
    required:true,
    trim:true,
  },

  subCategory: {
    type: String,
    required: true,
    trim: true,
  },

  offer : {
    type : String,
    trim : true,
  },

  
  price: {
    type: Number,
    // required: true,
    trim: true,
  },

  discountPrice: {
    type: Number,
    // required: true,
    trim: true,
  },
  
  images: [
    {
      type: String,
      required: true,
    },
  ],

  quantity: {
    type: Number,
    
  },

  // scheduleOptions : [
  //   {
  //     type : String,
  //     // required: true,
  //   }
  // ]

  schedule : {
    type : String,
  }

  // cart :[
  //   {
  //     product : this.productSchema
  //   }
  // ]

//   ratings: [ratingSchema],
});

const Product = mongoose.model("Product", productSchema);
module.exports = { Product,   productSchema };